server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -241,73 +239,38 @@ export class AssetService {
-    res: Res,
-    headers: Record<string, string>,
-  ) {
+  async getAssetThumbnail(authUser: AuthUserDto, assetId: string, query: GetAssetThumbnailDto, res: Res) {
     await this.access.requirePermission(authUser, Permission.ASSET_VIEW, assetId);
 
     const asset = await this._assetRepository.get(assetId);
     if (!asset) {
       throw new NotFoundException('Asset not found');
     }
 
-    try {
-      const thumbnailPath = this.getThumbnailPath(asset, query.format);
-      return this.streamFile(thumbnailPath, res, headers);
-    } catch (e) {
-      res.header('Cache-Control', 'none');
-      this.logger.error(`Cannot create read stream for asset ${asset.id}`, 'getAssetThumbnail');
-      throw new InternalServerErrorException(
-        `Cannot read thumbnail file for asset ${asset.id} - contact your administrator`,
-        { cause: e as Error },
-      );
-    }
+    const thumbnailPath = this.getThumbnailPath(asset, query.format);
+    this.streamFile(res, thumbnailPath);
   }
 
-  public async serveFile(
-    authUser: AuthUserDto,
-    assetId: string,
-    query: ServeFileDto,
-    res: Res,
-    headers: Record<string, string>,
-  ) {
+  public async serveFile(authUser: AuthUserDto, assetId: string, query: ServeFileDto, res: Res) {
     // this is not quite right as sometimes this returns the original still
     await this.access.requirePermission(authUser, Permission.ASSET_VIEW, assetId);
-
     const allowOriginalFile = !!(!authUser.isPublicUser || authUser.isAllowDownload);
 
     const asset = await this._assetRepository.getById(assetId);
     if (!asset) {
       throw new NotFoundException('Asset does not exist');
     }
 
-    // Handle Sending Images
-    if (asset.type == AssetType.IMAGE) {
-      try {
-        const { filepath, contentType } = this.getServePath(asset, query, allowOriginalFile);
-        return this.streamFile(filepath, res, headers, contentType);
-      } catch (e) {
-        this.logger.error(`Cannot create read stream for asset ${asset.id} ${JSON.stringify(e)}`, 'serveFile[IMAGE]');
-        throw new InternalServerErrorException(
-          e,
-          `Cannot read thumbnail file for asset ${asset.id} - contact your administrator`,
-        );
-      }
+    let filepath;
+    let contentType;
+    if (asset.type === AssetType.IMAGE) {
+      ({ filepath, contentType } = this.getServePath(asset, query, allowOriginalFile));
     } else {
-      try {
-        const videoPath = asset.encodedVideoPath ? asset.encodedVideoPath : asset.originalPath;
-        const mimeType = asset.encodedVideoPath ? 'video/mp4' : asset.mimeType;
-
-        return this.streamFile(videoPath, res, headers, mimeType);
-      } catch (e: Error | any) {
-        this.logger.error(`Error serving VIDEO asset=${asset.id}`, e?.stack);
-        throw new InternalServerErrorException(`Failed to serve video asset ${e}`, 'ServeFile');
-      }
+      filepath = asset.encodedVideoPath ? asset.encodedVideoPath : asset.originalPath;
```

Comment:
Maybe pull this out into a proper if statement given they're both evaluating the same condition?